docs/content/deployment.tex

```diff
@@ -0,0 +1,84 @@
+
+\subsection{Installation von Docker und docker-compose}
+Da sowohl der nginx-proxy, als auch das Front-und Backend mit Docker betrieben werden sollen, muss Docker und docker-compose installiert werden. Installationsanleitungen können entsprechend auf \cite{DockerInstallation} und \cite{DockerDomposeInstallation} gefunden werden.
+
+Um die Installation zu vereinfachen kann auch das Skript unter \lstinline{https://github.com/larsrickert/nginx-proxy/blob/main/utils/install-docker-ubuntu.sh} verwendet werden. Es beinhaltet die offiziellen Installationsschritte zur Installation der aktuellsten Versionen von Docker und docker-compose.
+
+Dazu muss die genannte Datei auf den \ac{VPS} kopiert werden und mit \lstinline{bash ./install-docker-ubuntu.sh} ausgeführt werden.
+
+\subsection{Installation des nginx-proxy}
+Nach der Installation von Docker und docker-compose muss der nginx-proxy selbst installiert werden. Dazu kann die Installationsanleitung unter \cite{NginxProxyInstallation} (\lstinline{https://github.com/larsrickert/nginx-proxy}) verwendet werden.
+
+Der nginx-proxy ist dafür zuständig, jede Docker-Anwendung, die die Umgebungsvariablen \lstinline{VIRTUAL_HOST} und \lstinline{LETSENCRYPT_HOST} gesetzt hat, über ebendiesen host/Domain zur Verfügung zu stellen. Dabei verwaltet er automatisch SSL-Zertifikate von Let's Encrypt, um die Verbindung über SSL zu verschlüsseln \cite{NginxProxyInstallation}.
+
+\section{Starten des Front- und Backends}
+Um nun das Front- und Backend dieser Studienarbeit bereitzustellen, muss zuerst mit \lstinline{git clone https://github.com/philippabele/nursing-home-volunteer.git} der Code von GitHub heruntergeladen werden.
+
+Danach muss die Datei \lstinline{.env.example} kopiert und in \lstinline{.env} umbenannt werden. Diese Datei enthält alle sensiblen Umgebungsvariablen für den Betrieb der beiden Anwendungen. Diese sollten niemals öffentlich geteilt werden. Nach dem Anlegen der \lstinline{.env} Datei sollten die Werte der Umgebungsvariablen entsprechend geändert werden.
+
+Nun kann das Front- und Backend durch \lstinline{docker-compose up -d} gestartet werden. Der zuvor eingerichtete nginx-proxy erkennt die beiden Anwendungen automatisch und stellt sie entsprechend über die \lstinline{FRONTEND_DOMAIN} und \lstinline{BACKEND_DOMAIN} bereit, die in der \lstinline{.env} Datei eingetragen wurden. Es sei angemerkt, dass die Anforderung der SSL-Zertifikate für die Domains ein paar Sekunden dauern kann, sodass direkt nach dem Starten eine Fehlermeldung des Browser erscheinen kann, dass die SSL-Zertifikate nicht gültig sind.
+
+Sollten neue Änderungen für die Anwendungen verfügbar, d.h. der Code auf GitHub geändert worden sein, können sie mit \lstinline{git pull && docker-compose up -d --build} aktualisiert werden.
```

Comment:
Please provide commands not as inline :)

The best technical instructions are organized like listings.

In best case, a future experienced reader only needs your commands to understand the story. The continuous text around your commands is like a frame for unexperienced readers.

Reply:
Sure, I changed it!